### PR TITLE
[Mobile app] data serialisation and map hot fixes.

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)

- [x] Resolves date time serialisation bugs.
- [x] Adds check for pm2.5 null values.
- [x] Updates iOS google maps and partly resolves [this firebase issue](https://console.firebase.google.com/u/0/project/airqo-250220/crashlytics/app/android:com.airqo.app/issues/66d1c36f11f0c6fbe16cff358125911d?time=last-seven-days&sessionEventKey=63C79BC50072000E567099EE4DACE828_1768332253145358303)

#### Status of maturity (all need to be checked before merging):

- [x] I've tested this locally
- [x] I consider this code done
- [x] This change ready to hit production in its current state
- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### What are the relevant tickets?
- [MOB-160](https://airqoteam.atlassian.net/browse/MOB-160)

#### Screenshots (optional)

[MOB-160]: https://airqoteam.atlassian.net/browse/MOB-160?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mobile/ios/Podfile.lock [hunk 1]
```diff
@@ -779,29 +779,29 @@ PODS:
...
     - nanopb (< 2.30910.0, >= 2.30908.0)
     - PromisesObjC (< 3.0, >= 1.2)
-  - GoogleMaps (7.2.0):
-    - GoogleMaps/Maps (= 7.2.0)
-  - GoogleMaps/Base (7.2.0)
-  - GoogleMaps/Maps (7.2.0):
+  - GoogleMaps (7.3.0):
+    - GoogleMaps/Maps (= 7.3.0)
+  - GoogleMaps/Base (7.3.0)
+  - GoogleMaps/Maps (7.3.0):
```
Comment (NoahNsimbe) [inline mobile/ios/Podfile.lock:785]: Updating google maps for iOS

File: mobile/lib/models/air_quality_reading.dart [hunk 2]
```diff
@@ -143,11 +146,11 @@ class AirQualityReading extends HiveObject {
...
   final DateTime dateTime;
 
-  @HiveField(9, defaultValue: 0.0)
+  @HiveField(9)
   final double pm2_5;
 
-  @HiveField(10, defaultValue: 0.0)
-  final double pm10;
+  @HiveField(10)
+  final double? pm10;
```
Comment (NoahNsimbe) [inline mobile/lib/models/air_quality_reading.dart:153]: made pm10 optional since it is not used

File: mobile/lib/utils/extensions.dart [hunk 3]
```diff
@@ -265,6 +265,28 @@ extension ProfileExt on Profile {
...
+    if (isYesterday()) {
+      return 'Updated yesterday at $dateString';
+    } else if (isToday()) {
+      return 'Updated today at $dateString';
+    } else if (isTomorrow()) {
+      return 'Tomorrow, $dateString';
+    } else {
+      return DateFormat('d MMM, hh:mm a').format(this);
+    }
+  }
```
Comment (NoahNsimbe) [inline mobile/lib/utils/extensions.dart:283]: last updated formatting

File: mobile/lib/models/air_quality_reading.dart [hunk 4]
```diff
@@ -27,19 +28,21 @@ class AirQualityReading extends HiveObject {
...
+    dateTime = dateTime.add(Duration(hours: DateTime.now().getUtcOffset()));
     PollutantValue pm2_5 =
         PollutantValue.fromJson(json["pm2_5"] as Map<String, dynamic>);
     PollutantValue pm10 =
         PollutantValue.fromJson(json["pm10"] as Map<String, dynamic>);
 
+    if (pm2_5.displayValue() == null) {
+      throw Exception("pm2.5 is null for site ${json["siteDetails"]["_id"]}");
+    }
+
```
Comment (NoahNsimbe) [inline mobile/lib/models/air_quality_reading.dart:40]: Raising exception incase pm2.5 is null

File: mobile/lib/screens/dashboard/dashboard_view.dart [hunk 5]
```diff
@@ -348,7 +348,7 @@ class _DashboardViewState extends State<DashboardView>
     );
   }
 
-  void _refresh({bool refreshMap = true}) {
+  Future<void> _refresh({bool refreshMap = true}) async {
```
Comment (BenjaminSsempala) [inline mobile/lib/screens/dashboard/dashboard_view.dart:351]: @NoahNsimbe I noticed this dart metric warning. Does removing the async cause errors?

**Review by Mnoble-19 — APPROVED** [review]
Tested and it's working fine on iOS. 
Just similar inquiries as @BenjaminSsempala

**Review by BenjaminSsempala — APPROVED** [review]
Tested out on Android. App works as expected.

File: mobile/lib/screens/dashboard/dashboard_view.dart (hunk 5 above)
Comment (NoahNsimbe) [reply mobile/lib/screens/dashboard/dashboard_view.dart]: Thanks @BenjaminSsempala  @Mnoble-19 I have resolved it.

**Pull request merged**